src/transformers/models/ctrl/modeling_ctrl.py

```diff
@@ -796,7 +796,9 @@ def forward(
             sequence_lengths = -1
         else:
             if input_ids is not None:
-                sequence_lengths = (torch.eq(input_ids, self.config.pad_token_id).int().argmax(-1) - 1).to(
+                # if no pad token found, use input_ids.shape[-1] - 1 instead of reverse indexing for ONNX compatibility
+                sequence_lengths = torch.eq(input_ids, self.config.pad_token_id).int().argmax(-1) - 1
+                sequence_lengths = torch.where(sequence_lengths >= 0, sequence_lengths, input_ids.shape[-1] - 1).to(
                     logits.device
                 )
```

Comment:
* Let's split across lines - it'll use fewer lines overall
* Can we instead use modulo to convert to the equivalent negative index? The logic at the moment assumes we only ever want to take the final index and I think it'll be faster than torch.where 

```suggestion
                sequence_lengths = sequence_lengths % input_ids.shape[-1]
                sequence_lengths = sequence_lengths.to(logits.device)
```

Reply:
That should work. Updated all changes